### PR TITLE
New Presence: ZombsRoyale.io

### PR DESCRIPTION
**Pull request by MrBartusek**
Fixes: #3680

image
image
image

### CODE REVIEW

**Comment by MrBartusek** [comment]
False-positive from codacy

File: websites/Z/ZombsRoyale.io/presence.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  clientId: "845354103118364672"
+});
+
+let matchStart: number = null;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  currentGameState = await this.getPageletiable("game.currentGameState");
```
Comment (ririxidev) [inline websites/Z/ZombsRoyale.io/presence.ts:11]: yeah well it isn't loool. it should be
```suggestion
  currentGameState = await getPageletiable("game.currentGameState");
```

**Review by Bas950 — DISMISSED** [review]

**Review by ririxidev — CHANGES_REQUESTED** [review]
isn't there more to show instead of only `viewing, buying` etc? also check type in ifs too
`==` -> `===`

**Comment by MrBartusek** [comment]
> isn't there more to show instead of only `viewing, buying` etc? also check type in ifs too
> `==` -> `===`

I searched a bit but it seems like there isn't anything else that `game.currentGameState`

File: websites/Z/ZombsRoyale.io/presence.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  const playing = presenceData.details === "In Game" || presenceData.details === "In Lobby";
+
+  if(playing && matchStart === null)
+  {
+    matchStart = Math.floor(Date.now() / 1000);
+  }
+  else if(!playing && matchStart !== null)
+  {
+    matchStart = null;
+  }
```
Comment (ririxidev) [inline websites/Z/ZombsRoyale.io/presence.ts:71]: [Start a Codeblock's Curly Braces on the Same Line](https://github.com/goldbergyoni/nodebestpractices#-33-start-a-codeblocks-curly-braces-on-the-same-line)

File: websites/Z/ZombsRoyale.io/presence.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    presenceData.state = "Parachuting";
+  } else if (currentGameState === "Dead" || currentGameState === "UiGameOver" || currentGameState === "UiSpectator" || currentGameState === "UiFeedbackOverlay") {
+    presenceData.details = "In Game";
+    presenceData.state = "Dead";
+  } else if (currentGameState === "UiMapOverlay") {
+    presenceData.details = "In Game";
+    presenceData.state = "Viewing map";
+  }else {
+    presence.error("Unknown state: " + currentGameState);
+  }
```
Comment (ririxidev) [inline websites/Z/ZombsRoyale.io/presence.ts:60]:
```suggestion
  } else presence.error("Unknown state: " + currentGameState);
```

File: websites/Z/ZombsRoyale.io/presence.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    presenceData.details = "Buying a Battle Pass";
+  } else if (currentGameState.startsWith("UiSeason") || currentGameState === "UiPreviewItemOverlay") {
+    presenceData.details = "Viewing Battle Pass";
+  } else if (currentGameState === "UiChallengesOverlay") {
+    presenceData.details = "Viewing challenges";
+  } else if (currentGameState === "UiBuyGemsOverlay") {
+    presenceData.details = "Buying gems";
+  } else if (currentGameState === "VideoAd") {
+    presenceData.details = "Watching video ad";
+  } else if (currentGameState === "Countdown" || currentGameState === "Lobby") {
```
Comment (ririxidev) [inline websites/Z/ZombsRoyale.io/presence.ts:41]:
```suggestion
  if (currentGameState === "loading" || currentGameState === "UiLoadingOverlay")
    presenceData.details = "Loading...";
  else if (currentGameState === "MainMenu")
    presenceData.details = "Lurking in main menu";
  else if (currentGameState === "Cosmetics" || currentGameState === "UiCosmeticSelectorOverlay")
    presenceData.details = "Viewing cosmetics";
  else if (currentGameState === "Profile")
    presenceData.details = "Viewing their profile";
  else if (currentGameState === "Shop" || currentGameState === "UiPreviewPackOverlay")
    presenceData.details = "Browsing Shop";
  else if (currentGameState === "Friends" || currentGameState === "UiFriendAddOverlay")
    presenceData.details = "Viewing their friends";
  else if (currentGameState === "Leaderboards")
    presenceData.details = "Browsing leaderboards";
  else if (currentGameState === "UiLoginOverlay")
    presenceData.details = "Loging in";
  else if (currentGameState === "UiSettingsOverlay")
    presenceData.details = "Changing settings";
  else if (currentGameState === "UiSeasonPurchaseOverlay")
    presenceData.details = "Buying a Battle Pass";
  else if (currentGameState.startsWith("UiSeason") || currentGameState === "UiPreviewItemOverlay")
    presenceData.details = "Viewing Battle Pass";
  else if (currentGameState === "UiChallengesOverlay")
    presenceData.details = "Viewing challenges";
  else if (currentGameState === "UiBuyGemsOverlay")
    presenceData.details = "Buying gems";
  else if (currentGameState === "VideoAd")
    presenceData.details = "Watching video ad";
  else if (currentGameState === "Countdown" || currentGameState === "Lobby") {
```

**Review by ririxidev — CHANGES_REQUESTED** [review]
and run prettier over it

File: websites/Z/ZombsRoyale.io/presence.ts (hunk 4 above)
Comment (MrBartusek) [reply websites/Z/ZombsRoyale.io/presence.ts:41]: Not sure what changed?

Comment (ririxidev) [reply websites/Z/ZombsRoyale.io/presence.ts:41]: nothing. code style 😛

File: websites/Z/ZombsRoyale.io/presence.ts (hunk 3 above)
Comment (MrBartusek) [reply websites/Z/ZombsRoyale.io/presence.ts:60]: `presence.error` is doubled now?

Comment (ririxidev) [reply websites/Z/ZombsRoyale.io/presence.ts:60]: how that's doubled

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**